receiver/pulsarreceiver/receiver_traces.go

```diff
@@ -0,0 +1,113 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package pulsarreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/pulsarreceiver"
+import (
+	"context"
+	"errors"
+	"strings"
+	"time"
+
+	"github.com/apache/pulsar-client-go/pulsar"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/receiver"
+	"go.uber.org/zap"
+)
+
+type pulsarTracesReceiver struct {
+	tracesConsumer consumer.Traces
+	client         pulsar.Client
+	cancel         context.CancelFunc
+	consumer       pulsar.Consumer
+	unmarshaler    TracesUnmarshaler
+	settings       receiver.CreateSettings
+	config         Config
+}
+
+func newTracesReceiver(config Config, set receiver.CreateSettings, unmarshalers map[string]TracesUnmarshaler, nextConsumer consumer.Traces) (*pulsarTracesReceiver, error) {
+	option := config.Trace
+	if err := option.validate(); err != nil {
+		return nil, err
+	}
+	unmarshaler := unmarshalers[option.Encoding]
+	if nil == unmarshaler {
+		return nil, errUnrecognizedEncoding
+	}
+
+	return &pulsarTracesReceiver{
+		tracesConsumer: nextConsumer,
+		unmarshaler:    unmarshaler,
+		settings:       set,
+		config:         config,
+	}, nil
+}
+
+func (c *pulsarTracesReceiver) Start(context.Context, component.Host) error {
+	client, _consumer, err := c.config.createConsumer(c.config.Log)
+	if err != nil {
+		return err
+	}
+	ctx, cancel := context.WithCancel(context.Background())
+	c.cancel = cancel
+	c.client = client
+	c.consumer = _consumer
+	go func() {
+		if e := consumerTracesLoop(ctx, c); e != nil {
+			c.settings.Logger.Error("consume traces loop occurs an error", zap.Error(e))
+		}
+	}()
+	return err
+}
+
+func consumerTracesLoop(ctx context.Context, c *pulsarTracesReceiver) error {
```

Comment:
```suggestion
func consumeTracesLoop(ctx context.Context, c *pulsarTracesReceiver) error {
```

Just to be consistent with `consumeMetricsLoop` and `consumeLogsLoop`